cluster/addons/addon-manager/kube-addons.sh

```diff
@@ -275,20 +275,14 @@ function is_leader() {
   # Disabling because "${KUBECTL_OPTS}" needs to allow for expansion here
   KUBE_CONTROLLER_MANAGER_LEADER=$(${KUBECTL} ${KUBECTL_OPTS} -n kube-system get leases.v1.coordination.k8s.io kube-controller-manager -o "jsonpath={.spec.holderIdentity}")
 
-  case "${KUBE_CONTROLLER_MANAGER_LEADER}" in
-  "")
+  if [ -z "${KUBE_CONTROLLER_MANAGER_LEADER}" ]; then
     log ERR "No leader election info found."
     return 1
-    ;;
-
-  "${HOSTNAME}"_*)
+  elif [[ "${KUBE_CONTROLLER_MANAGER_LEADER}" =~ ^${HOSTNAME}_[a-zA-Z0-9-] || "${KUBE_CONTROLLER_MANAGER_LEADER}" =~ ^${HOSTNAME}---[a-zA-Z0-9-] ]]; then
```

Comment:
I'd be okay with leaving the legacy format as-is (wildcard match).  If we do change it, we need to be very confident this always works.  Regex on new format looks good.

Reply:
I would prefer keeping the legacy format as well (and maybe work on improving the regex as a followup), we don't have existing tests and this code path is only exercised in pull-kubernetes-e2e-gce.